torch/csrc/autograd/python_engine.cpp

```diff
@@ -80,11 +80,12 @@ void PythonEngine::thread_init(int device, const std::shared_ptr<ReadyQueue>& re
     decrement_non_reentrant_thread_count();
   }
 
-#ifdef IS_PYTHON_3_9_PLUS
+#if defined(IS_PYTHON_3_9_PLUS) || defined(USE_DEPLOY)
```

Comment:
Not sure I understand this. The problem as described in the PR summary is about autograd's behavior on shutdown, but here it seems we're enabling logic in `thread_init`—how does it affect shutdown?

Reply:
@d1jang explained to me offline.

This looks good to me providing @albanD gets satisfactory answers to his questions